app/auth/error/page.tsx

```diff
@@ -0,0 +1,35 @@
+"use client";
+
+import { Fragment } from "react";
+
+import { signIn } from "next-auth/react";
+
+/** Custom sign-in error page, see documentation at https://next-auth.js.org/configuration/pages */
+const Page = () => {
+  const params = new URLSearchParams(
```

Comment:
Tohle ještě bude chtít nějakou lásku. Základní idea je jednoduše měnit chování stránky podle GET parametru `error`. Původně jsem tady k tomu měl hook `useSearchParams`, ale pak si stěžoval Next:

```
 ⨯ useSearchParams() should be wrapped in a suspense boundary at page "/auth/error". Read more: https://nextjs.org/docs/messages/missing-suspense-with-csr-bailout
```

Celkově moc nechápu, co po mně chce – takže jsem to jen naslepo nahradil tímhle řešením, které ale zase způsobuje chyby v hydrataci, protože verze renderovaná na serveru (bez parametrů) je jiná než verze oživlá na klientovi (vidí GET parametry).

Reply:
Jáj, to je teda pakárna, s tím `useEffect` je to zbytečná magie. Škoda. Díky!